File: frontend/app/src/main/java/com/example/geoquiz_frontend/data/remote/dtos/question/QuestionData.java
package com.example.geoquiz_frontend.data.remote.dtos.question;

import com.example.geoquiz_frontend.domain.enums.LocalizedText;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class QuestionData {

    @SerializedName("countryId")
    private String countryId;

    @SerializedName("questionText")
    private LocalizedText questionText;

    @SerializedName("options")
    private List<OptionData> options;

    @SerializedName("imageUrl")
    private String imageUrl;

    @SerializedName("audioUrl")
    private String audioUrl;

    public String getCountryId() { return countryId; }
    public LocalizedText getQuestionText() { return questionText; }
    public List<OptionData> getOptions() { return options; }
    public String getImageUrl() { return imageUrl; }
    public String getAudioUrl() { return audioUrl; }
}

File: backend/GeoQuiz_backend/GeoQuiz_backend/Application/Payloads/Questions/QuestionData.cs
using GeoQuiz_backend.Domain.Mongo;

namespace GeoQuiz_backend.Application.Payloads.Questions
{
    public class QuestionData
    {
        public string CountryId { get; set; } = null!;
        public LocalizedText QuestionText { get; set; } = null!;
        public List<GameOption> Options { get; set; } = new();
        public string? ImageUrl { get; set; }
        public string? AudioUrl { get; set; }
    }
    public class GameOption
    {
        public int Index { get; set; }
        public LocalizedText Text { get; set; } = null!;
    }
}
